tests/gateway/providers/test_openai.py

```diff
@@ -290,4 +323,178 @@ async def test_embeddings_batch_input():
+    [
+        # Invalid API Types
+        ("invalidtype", None, None, None, None, MlflowException),
+        (0, None, None, None, None, MlflowException),
+        # OpenAI API type
+        ("openai", None, None, None, None, None),
+        ("openai", "https://api.openai.com/v1", None, None, None, None),
+        ("openai", "https://api.openai.com/v1", None, "2023-05-15", None, None),
+        ("openAI", "https://api.openai.com/v1", None, "2023-05-15", None, None),
+        ("openAI", "https://api.openai.com/v1", None, "2023-05-15", "test-organization", None),
+        # Deployment name is specified when API type is not 'azure' or 'azuread'
+        ("openai", None, "mock-deployment", None, None, MlflowException),
+        # Azure API type
+        ("azure", "https://test.openai.azure.com", "mock-dep", "2023-05-15", None, None),
+        ("AZURe", "https://test.openai.azure.com", "mock-dep", "2023-04-12", None, None),
+        # Missing required API base, deployment name, and / or api version fields
+        ("azure", None, None, None, None, MlflowException),
+        ("azure", "https://test.openai.azure.com", "mock-dep", None, None, MlflowException),
+        # Organization is specified when API type is not 'openai'
+        ("azure", "https://test.openai.azure.com", "mock-dep", "2023", "test-org", MlflowException),
+        # AzureAD API type
+        ("azuread", "https://test.openai.azure.com", "mock-dep", "2023-05-15", None, None),
+        ("azureAD", "https://test.openai.azure.com", "mock-dep", "2023-04-12", None, None),
+        # Missing required API base, deployment name, and / or api version fields
+        ("azuread", None, None, None, None, MlflowException),
+        ("azuread", "https://test.openai.azure.com", "mock-dep", None, None, MlflowException),
+        # Organization is specified when API type is not 'openai'
+        ("azuread", "https://test.openai.azure.com", "mock", "2023", "test-org", MlflowException),
+    ],
+)
+def test_openai_provider_validates_openai_config(
+    api_type,
+    api_base,
+    deployment_name,
+    api_version,
+    organization,
+    exception_type,
+):
+    openai_config = OpenAIConfig(
+        openai_api_key="mock-api-key",
+        openai_api_type=api_type,
+        openai_api_base=api_base,
+        openai_deployment_name=deployment_name,
+        openai_api_version=api_version,
+        openai_organization=organization,
+    )
+    route_config = RouteConfig(
+        name="completions",
+        route_type="llm/v1/completions",
+        model={
+            "provider": "openai",
+            "name": "text-davinci-003",
+            "config": dict(openai_config),
+        },
+    )
+    if exception_type is not None:
+        with pytest.raises(exception_type, match="OpenAI"):
+            OpenAIProvider(route_config)
+    else:
+        provider = OpenAIProvider(route_config)
+        assert provider.openai_config == openai_config
```

Comment:
/ai split this test into two, one for ok cases, one for bad cases.

Reply:
Good idea - I've spit these up